src/main/java/httpserver/HTTPRequest.java

```diff
@@ -25,509 +25,502 @@
  * @see HTTPHandler
  */
 public class HTTPRequest implements Runnable {
```

Comment:
We should consider getting rid of all of communication layer stuff, and just transition to having a simplifying layer on top of something more robust, like Jetty or Tomcat.

My thinking, for the most part, is that they probably did it better, so why should we try to compete with them. Where we can excel is as a simplified layer.